br/pkg/lightning/config/config.go

```diff
@@ -1024,6 +1024,10 @@ func (cfg *Config) CheckAndAdjustTiDBPort(ctx context.Context, mustHaveInternalC
 	if cfg.TiDB.Port <= 0 {
 		return errors.New("invalid `tidb.port` setting")
 	}
+	// remove the http(s) schema and we will fill it depend on the tls settings.
+	cfg.TiDB.PdAddr = strings.TrimPrefix(cfg.TiDB.PdAddr, "https://")
+	cfg.TiDB.PdAddr = strings.TrimPrefix(cfg.TiDB.PdAddr, "http://")
```

Comment:
It seems we always use the host part of `PdAddr` or assume it is a host. I think it is more appropriate to use `url.Parse` to parse its host.

Reply:
removed as this is not a complete solution. We may fully resolve this issue in a separate PR.